go-controller/pkg/metrics/ovn.go

```diff
@@ -342,6 +349,65 @@ func getPortCount(portType string) float64 {
 	return portCount
 }
 
+//ovnControllerSBDBConnectionCheckUpdater blocks until stopCh closed but before then polls ovn-controllers connection status with
+//southbound database periodically.
+func ovnControllerSBDBConnectionCheckUpdater(stopCh <-chan struct{}, ovsAppctl ovsClient, period time.Duration) {
+	// There maybe transient connection issues to SB DB. We want to minimise the risk of reporting this as the current state between
+	// long poll intervals.
+	retry := 5
+	sleepBetweenRetry := 5 * time.Second
+	// retry times sleepBetweenRetry
+	totalRetriesTime := 25 * time.Second
+
+	if totalRetriesTime.Seconds() >= period.Seconds() {
+		panic("period must be greater than retry total time")
+	}
+	// update metric to a good initial state
+	updateSBDBConnectionMetric(ovsAppctl, retry, sleepBetweenRetry)
+
+	ticker := time.NewTicker(period)
```

Comment:
`defer ticker.Stop()` reduces my anxiety

Reply:
haha! Do you really want me to change the code for that?

Reply:
I think using defer in go where possible is a good style (but I will give you lgtm anyway :D)

Reply:
true.. if its reworked in the future, someone could make a mistake, and we know theres no cost to using defer now with the go version we use.